internal/or/or.go

```diff
@@ -0,0 +1,13 @@
+// Package or helps with a simple use case: defining fallback values, similar
+// to the null coalescing operator.
+package or
```

Comment:
Perhaps something like stringUtils? Seeing only `or` in a directory listing is unlikely to come with enough context that it could be considered a descriptive name.

Reply:
Want the name to also be short, and this logic could easily be extended for other types so I don't think adding `string` into the package name is needed.

What about `nonzero.String`?

Reply:
More alternatives:
`first.String`
`valid.String`: Feels more like it would return a bool, don't particularly like it myself.
`fallback.String`: A bit long but fine I think.

I prefer `first`, `or`, and `fallback`.

Reply:
More alternatives:
- `firstnonzero.String`
- `nonzero.FirstString`

Reply:
In SQL it's called `coalesce`

Reply:
@Pikabanga Did not know that! I'm changing to that until someone complaints

Reply:
Noone complained. I'm merging.